tools/LogCollector/source/omiHighCPUDiagnostics.sh

```diff
@@ -0,0 +1,205 @@
+#!/bin/bash
+#set -x
+declare -a ppuc
+declare -a pt
+count=2880
+sleep_time_btw_each_collection=5 #seconds
+
+if [ "$1" = "--help" ]; then
+    echo "This scipt runs for 24 hour and collect info about omiagent process with high cpu utilization(>20%)."
+    echo "When CPU utilization is high it takes the stack trace snapshot for the omiagent for 10 times in interval of 10 sec."
+    echo "Then rescan the system in interval of 10sec."
+    echo "It also collects the diff of scx and omi logs in every 10 sec"
+    echo "  Option            Description"
+    echo "--runtime-in-min    set run time in minutes, default 3600 -> 24 hours"
+    echo "--trace-enable      set verbose logging for omi and scx with option trace-enable"
+    echo "--trace-disable     disable verbose logging for omi and scx"
+    exit 0;
+
+elif [ "$1" = "--trace-enable" ]; then
+    /opt/microsoft/scx/bin/tools/scxadmin -log-set all intermediate
+    sed -i 's/#*loglevel/loglevel/' /etc/opt/omi/conf/omiserver.conf
+    sed -i 's/WARNING/DEBUG/' /etc/opt/omi/conf/omiserver.conf
+    /opt/microsoft/scx/bin/tools/scxadmin -restart
+    exit 0
+
+elif [ "$1" = "--trace-disable" ]; then
```

Comment:
Not guaranteed that the customer is using levels "-log-set all errors" before the script is run (--trace-enable could clobber existing settings). Maybe upon --trace-enable you can check if existing settings are non-default and give a warning.

Reply:
done